### PR TITLE
Persists user information during role entry and publication requests

### PR DESCRIPTION
**Pull request by pumazi**
Depends on Connexions/openstax-accounts#16 & Connexions/cnx-archive#290

See also [trello card](https://trello.com/c/2Z2OEcSq/1116-convert-users-table-from-fdw-to-shadow-table)

### CODE REVIEW

**Comment by pumazi** [comment]
Looks like a release of openstax-accounts is going to be required.

**Comment by pumazi** [comment]
The coverage is not going to pass anytime soon. I'm managed to scope the coverage report, but that has brought down the coverage by nearly 2%. The good thing is that it is now more accurate.

File: cnxpublishing/db.py [hunk 1]
```diff
@@ -1129,6 +1135,67 @@ def remove_acl(cursor, uuid_, permissions):
...
+                   (user_ids,))
+    try:
+        existing_user_ids = [x[0] for x in cursor.fetchall()]
+    except TypeError:
+        existing_user_ids = []
+    new_user_ids = [u for u in user_ids if u not in existing_user_ids]
+
+    # Update existing records.
+    for user_id in existing_user_ids:
+        # TODO only update based on a delta against the 'updated' column.
```
Comment (reedstrm) [inline cnxpublishing/db.py]: doesn't really save anything: postgresql will need to form an entire new tuple anyway, and the data here is so small not sending the unchanged values isn't worth it. Simple is better. So, no TODO

File: cnxpublishing/tests/test_views.py [hunk 2]
```diff
@@ -818,6 +828,21 @@ def test_new_to_publication(self):
...
+            with db_conn.cursor() as cursor:
+                cursor.execute("SELECT username, "
+                               "ARRAY[first_name IS NOT NULL, "
+                               "      last_name IS NOT NULL, "
+                               "      full_name IS NOT NULL] "
+                               "FROM users ORDER BY username")
+                users = {u:set(b) for u, b in cursor.fetchall()}
+        self.assertEqual(sorted(users.keys()), sorted(uids))
+        for username, null_checks in users.items():
+            self.assertNotIn(None, null_checks,
```
Comment (reedstrm) [inline cnxpublishing/tests/test_views.py]: clever :+1:

**Pull request merged**